cdap-api/src/main/java/co/cask/cdap/api/dataset/lib/IndexedObjectStoreDefinition.java

```diff
@@ -32,53 +30,21 @@
  */
 @Beta
 public class IndexedObjectStoreDefinition
-  extends AbstractDatasetDefinition<IndexedObjectStore, DatasetAdmin> {
-
-  private final DatasetDefinition<? extends Table, ?> tableDef;
-  private final DatasetDefinition<? extends ObjectStore, ?> objectStoreDef;
+  extends CompositeDatasetDefinition<IndexedObjectStore> {
 
   public IndexedObjectStoreDefinition(String name,
                                       DatasetDefinition<? extends Table, ?> tableDef,
                                       DatasetDefinition<? extends ObjectStore, ?> objectStoreDef) {
-    super(name);
-    if (tableDef == null) {
-      throw new IllegalArgumentException("Table definition is required");
-    }
-    if (objectStoreDef == null) {
-      throw new IllegalArgumentException("ObjectStore definition is required");
-    }
-    this.tableDef = tableDef;
-    this.objectStoreDef = objectStoreDef;
-  }
-
-  @Override
-  public DatasetSpecification configure(String instanceName, DatasetProperties properties) {
-    return DatasetSpecification.builder(instanceName, getName())
-      .properties(properties.getProperties())
-      .datasets(tableDef.configure("index", properties),
-                objectStoreDef.configure("data", properties))
-      .build();
-  }
-
-  @Override
-  public DatasetAdmin getAdmin(DatasetContext datasetContext, DatasetSpecification spec,
-                               ClassLoader classLoader) throws IOException {
-    return new CompositeDatasetAdmin(
-      tableDef.getAdmin(datasetContext, spec.getSpecification("index"), classLoader),
-      objectStoreDef.getAdmin(datasetContext, spec.getSpecification("data"), classLoader)
-    );
+    super(name, "index", tableDef, "data", objectStoreDef);
   }
 
   @Override
   public IndexedObjectStore<?> getDataset(DatasetContext datasetContext, DatasetSpecification spec,
                                           Map<String, String> arguments, ClassLoader classLoader) throws IOException {
-    DatasetSpecification tableSpec = spec.getSpecification("index");
-    DatasetSpecification objectStoreSpec = spec.getSpecification("data");
 
-    Table index = tableDef.getDataset(datasetContext, tableSpec, arguments, classLoader);
-    ObjectStore<?> objectStore = objectStoreDef.getDataset(datasetContext, objectStoreSpec, arguments, classLoader);
+    Table index = super.getDataset(datasetContext, "index", spec, arguments, classLoader);
```

Comment:
Aahhh. This is a leftover. I tried to make it a nested expression, had to call <Table>getDataset(...) and you can only do that syntactically if you prefix it with this. or super. Changed it back and forgot to remove the super. Gone now. :)